src/lib/utils/calculatePagination.js

```diff
@@ -0,0 +1,44 @@
+import { ITEMS_IN_PAGINATION } from "./constants"
+
+function getBreakPoint(num) {
+  return Math.ceil(ITEMS_IN_PAGINATION * num)
+}
+
+export function calculatePagination({ page, pages, to, from }) {
```

Comment:
Wydaje mi sie, ze mozna by to mocno uproscic, wiec moze sprobuj jeszcze nad tym pomyslec